src/frontend/src/handler/query.rs

```diff
@@ -462,14 +473,26 @@ async fn distribute_execute(
 }
 
 #[expect(clippy::unused_async)]
-async fn local_execute(session: Arc<SessionImpl>, query: Query) -> Result<LocalQueryStream> {
+async fn local_execute(
+    session: Arc<SessionImpl>,
+    query: Query,
+    can_timeout_cancel: bool,
+) -> Result<LocalQueryStream> {
+    let timeout = if cfg!(madsim) {
```

Comment:
There is a bug in madsim which may lead to invalid memory access. We skip madsim using this approach for now.